File: examples/search_cars.py
"""
Search for cars on Sauto.cz with filters.

Usage:
    pip install requests
    python search_cars.py

Get your API key at: https://rapidapi.com/XrayHunter-F8lHMI9me/api/sauto-cz-czech-cars-api
"""

import requests

API_KEY = "YOUR_API_KEY"  # Replace with your RapidAPI key
BASE_URL = "https://sauto-cz-czech-cars-api.p.rapidapi.com"
HEADERS = {
    "X-RapidAPI-Key": API_KEY,
    "X-RapidAPI-Host": "sauto-cz-czech-cars-api.p.rapidapi.com",
}


def search(params: dict) -> dict:
    response = requests.get(f"{BASE_URL}/search", headers=HEADERS, params=params)
    response.raise_for_status()
    return response.json()


def main():
    # Example 1: Search for used Skoda Octavia diesel
    print("=== Used Skoda Octavia Diesel ===")
    data = search({
        "make": "skoda",
        "model": "octavia",
        "fuel": "diesel",
        "condition": "used",
        "per_page": 5,
    })
    print(f"Found {data['total_count']} cars\n")
    for car in data["results"]:
        print(f"  {car['make']} {car['model']} ({car['year_manufactured']})")
        print(f"    Price: {car['price']:,} CZK | Mileage: {car['mileage_km']:,} km")
        print(f"    Fuel: {car['fuel']} | Gearbox: {car['gearbox']}")
        print()

    # Example 2: Electric cars under 500K CZK
    print("=== Electric Cars Under 500K CZK ===")
    data = search({
        "fuel": "electric",
        "price_max": 500000,
        "per_page": 5,
    })
    print(f"Found {data['total_count']} cars\n")
    for car in data["results"]:
        print(f"  {car['make']} {car['model']} ({car['year_manufactured']}) | {car['price']:,} CZK")

    print()

    # Example 3: BMW automatic 4x4 with low mileage
    print("=== BMW Automatic 4x4, Under 100K km ===")
    data = search({
        "make": "bmw",
        "gearbox": "automatic",
        "drive": "4x4",
        "mileage_max": 100000,
        "per_page": 5,
    })
    print(f"Found {data['total_count']} cars\n")
    for car in data["results"]:
        print(f"  {car['make']} {car['model']} ({car['year_manufactured']}) | {car['price']:,} CZK | {car['mileage_km']:,} km")


if __name__ == "__main__":
    main()

File: examples/search_cars.js
/**
 * Search for cars on Sauto.cz with filters (Node.js example).
 *
 * Usage:
 *   npm install axios
 *   node search_cars.js
 *
 * Get your API key at: https://rapidapi.com/XrayHunter-F8lHMI9me/api/sauto-cz-czech-cars-api
 */

const axios = require('axios');

const API_KEY = 'YOUR_API_KEY'; // Replace with your RapidAPI key
const BASE_URL = 'https://sauto-cz-czech-cars-api.p.rapidapi.com';
const HEADERS = {
    'X-RapidAPI-Key': API_KEY,
    'X-RapidAPI-Host': 'sauto-cz-czech-cars-api.p.rapidapi.com',
};

async function search(params) {
    const { data } = await axios.get(`${BASE_URL}/search`, { headers: HEADERS, params });
    return data;
}

async function getVehicle(id) {
    const { data } = await axios.get(`${BASE_URL}/vehicle/${id}`, { headers: HEADERS });
    return data;
}

async function main() {
    // Search for SUVs with automatic gearbox
    console.log('=== SUVs with Automatic Gearbox ===\n');
    const data = await search({
        gearbox: 'automatic',
        drive: '4x4',
        price_max: 800000,
        per_page: 5,
    });

    console.log(`Found ${data.total_count} cars\n`);
    for (const car of data.results) {
        console.log(`  ${car.make} ${car.model} (${car.year_manufactured})`);
        console.log(`    Price: ${car.price?.toLocaleString()} CZK | Mileage: ${car.mileage_km?.toLocaleString()} km`);
        console.log(`    Fuel: ${car.fuel} | Gearbox: ${car.gearbox}`);
        console.log();
    }

    // Get full detail for first result
    if (data.results.length > 0) {
        const id = data.results[0].id;
        console.log(`\n=== Full Detail for ID ${id} ===\n`);
        const detail = await getVehicle(id);

        console.log(`${detail.make} ${detail.model} (${detail.year_manufactured})`);
        console.log(`VIN: ${detail.vin}`);
        console.log(`Price: ${detail.pricing.price?.toLocaleString()} CZK`);
        console.log(`Engine: ${detail.engine_power_kw} kW (${detail.engine_power_hp} HP), ${detail.engine_volume_cc} cc`);
        console.log(`Equipment: ${detail.equipment.length} items`);
        console.log(`Photos: ${detail.images.length}`);
        console.log(`Seller: ${detail.seller.phone} (${detail.seller.type})`);
    }
}

main().catch(console.error);
